datafusion/src/datasource/object_store/local.rs

```diff
@@ -82,7 +80,7 @@ impl ObjectReader for LocalFileReader {
         &self,
         start: u64,
         length: usize,
-    ) -> Result<Box<dyn Read + Send + Sync>> {
+    ) -> Result<Box<dyn ReadSeek + Send + Sync>> {
```

Comment:
Seek is needed for parquet `read_metadata`, `ReadSeek` brought in because of dyn trait limitation